### ISSUE TITLE
hyperparameters for resnet50 + xent loss

### ISSUE THREAD

**Issue by 0-oo-0** [KaiyangZhou/deep-person-reid#41]
Hi, I use the default setting when I train resnet50 with cross-entropy-label-smooth loss on Market1501, i.e. -- max-epoch 60 -- stepsize 20 40, then I got  map 67.3, rank-1 85.2. Therefore, I try --max-epoch 180 --stepsize 60, then I got
![default](https://user-images.githubusercontent.com/33923638/43363635-ae10cafe-933b-11e8-8b65-b58300f4b857.png)
 
my results seem to be much worse than yours :
![default](https://user-images.githubusercontent.com/33923638/43363647-4d91c8f8-933c-11e8-9788-4d3a8976fab7.png)

thank you very much.

**Comment by KaiyangZhou** [comment]
You can try the following parameters
```
fixbase_epoch=6, fixbase_lr=0.0003, freeze_bn=False, max_epoch=60, optim='amsgrad', stepsize=[20, 40], test_batch=100, train_batch=32
```

For imagenet-pretrained models such as resnet50, you may first train the randomly initialized classifier for several epochs and then open all layers for finetuning. This can prevent the well trained mapping functions (i.e. lower layers) from being contaminated by the fresh classifier.

**Comment by 0-oo-0** [comment]
That is cool. Training the randomly initialized classifier for 6 epochs make sense, and I obtain
![default](https://user-images.githubusercontent.com/33923638/43450953-72342dee-94e6-11e8-931a-c502ce3a29b9.png)
Although a little worse than your results, I will try --max-epoch 180 to obtain better results.
thank you.

**Comment by KaiyangZhou** [comment]
this paper would help https://arxiv.org/abs/1611.05244

**Comment by DL-85** [comment]
@0-oo-0 Did you use the optimizer amsgrad? Why I got 85.7(68.6) while using the parameters above?

**Comment by 0-oo-0** [comment]
@DL-85 I use Adam with amsgrad=False

**Comment by KaiyangZhou** [comment]
for resnet50, using either adam or amsgrad and following the default param settings can always give you around 88% rank1 (don't forget to freeze the base for several epochs e.g. 10).

if you want a stronger baseline, you can do sth like adding a fc layer before softmax, but doing so will decrease the cross dataset performance as the features from the fc layer are too task-specific, in this case you might wanna use the pool5 features instead, which could bring some improvements
